url10fb6bd43e_guwere_P2PSearch_tgz-pJ8-MainJ8/src/Sender.java

```diff
@@ -0,0 +1,42 @@
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketException;
+import java.util.concurrent.BlockingQueue;
+
+public class Sender implements Runnable {
+    private DatagramSocket socket;
+    DatagramPacket message;
+    BlockingQueue sendQueue;
+
+    public Sender(BlockingQueue sQueue){
+        try {
+            socket = new DatagramSocket();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *socket*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).